src/generators/csharp/presets/JsonSerializerPreset.ts

```diff
@@ -0,0 +1,247 @@
+  instance.${patternPropertyName}.Add(propertyName, deserializedValue);
+  continue;
+}`;
+  });
+  return patternProperties.join('\n');
+}
+
+function renderUnmarshalAdditionalProperties(model: CommonModel, renderer: CSharpRenderer) {
+  if (model.additionalProperties === undefined) {
+    return ''; 
+  }
+  let additionalPropertyName = getUniquePropertyName(model, DefaultPropertyNames.additionalProperties);
+  additionalPropertyName = FormatHelpers.upperFirst(renderer.nameProperty(additionalPropertyName, model.additionalProperties));
+  const additionalPropertyType = renderer.renderType(model.additionalProperties);
+  return `if(instance.${additionalPropertyName} == null) { instance.${additionalPropertyName} = new Dictionary<string, ${additionalPropertyType}>(); }
+var deserializedValue = JsonSerializer.Deserialize<${additionalPropertyType}>(ref reader, options);
+instance.${additionalPropertyName}.Add(propertyName, deserializedValue);
+continue;`;
+}
+
+/**
+ * Render `unmarshal` function based on model
+ */
+function renderUnmarshal({ renderer, model }: {
+  renderer: CSharpRenderer,
+  model: CommonModel,
+}): string {
+  const formattedModelName = renderer.nameType(model.$id);
+  const unmarshalProperties = renderUnmarshalProperties(model, renderer);
+  const unmarshalPatternProperties = renderUnmarshalPatternProperties(model, renderer);
+  const unmarshalAdditionalProperties = renderUnmarshalAdditionalProperties(model, renderer);
+  return `public override ${formattedModelName} Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+{
+  if (reader.TokenType != JsonTokenType.StartObject)
+  {
+    throw new JsonException();
+  }
+
+  var instance = new ${formattedModelName}();
+  
+  while (reader.Read())
+  {
+    if (reader.TokenType == JsonTokenType.EndObject)
+    {
+      return instance;
+    }
+
+    // Get the key.
+    if (reader.TokenType != JsonTokenType.PropertyName)
+    {
+      throw new JsonException();
+    }
+
+    string propertyName = reader.GetString();
+${renderer.indent(unmarshalProperties, 4)}
+
+${renderer.indent(unmarshalPatternProperties, 4)}
+
+${renderer.indent(unmarshalAdditionalProperties, 4)}
+  }
+  
+  throw new JsonException();
+}`;
+} 
+
+/**
+ * Preset which adds `marshal` and `unmarshal` functions to class. 
+ * 
+ * @implements {CSharpPreset}
+ */
+export const CSHARP_JSON_SERIALIZER_PRESET: CSharpPreset = {
+  class: {
+    async self({ renderer, model}) {
+      renderer.addDependency('using System.Text.Json;');
+      renderer.addDependency('using System.Text.Json.Serialization;');
+      renderer.addDependency('using System.Text.RegularExpressions;');
+      
+      const formattedModelName = renderer.nameType(model.$id);
+      const unmarshal = renderUnmarshal({renderer, model});
+      const marshal = renderMarshal({renderer, model});
+
+      return `[JsonConverter(typeof(${formattedModelName}Converter))]
+${await renderer.defaultSelf()}
```

Comment:
You can use the content from previous preset:

```suggestion
${content}
```

but then you must add `content` in the arguments  of `self` function.

Reply:
Done 👍